plugin_tooling/src-lang/melnorme/lang/tooling/structure/AbstractStructureContainer.java

```diff
@@ -43,7 +46,23 @@ protected void _invariant() {
 		return children;
 	}
 	
-	/* -----------------  ----------------- */
+	public Indexable<StructureElement> cloneSubTree() {
+		ArrayList2<StructureElement> clonedElements = new ArrayList2<>(children.size());
+		for(StructureElement child : children) {
+			clonedElements.add(child.cloneTree());
+		}
+		return clonedElements;
+	}
+	
+	public List<StructureElement> flattenSubTree() {
+		List<StructureElement> flattenedElements = new ArrayList<>();
+		for(StructureElement child : getChildren()) {
+			flattenedElements.addAll(child.flattenTree());
+		}
+		return flattenedElements;

```

Comment:
Sounds reasonable. Was not so happy about the flatten neither. I will think about it.